### ISSUE TITLE
Performance difference for GEMM & flash-attention between reported and reproduced in CI

### ISSUE THREAD

**Issue by Egor-Krivov** [intel/intel-xpu-backend-for-triton#1470]
Current CI runs POC branches for flash-attention `perf_attn` and for GEMMs `triton_perf_poc` and runs provided `run_all.sh` scripts. Custom libigc is installed `libigc1_1.0.24994.16243-igc+releaseinternal1_amd64.deb`. However, there is a significant performance difference between reproduced and reported. 

For example in flash attention CI reproduction gets about 2.5TFlops for `Z:1,H:32,N_CTX:16384,D_HEAD:64`, while reported value is close to 81 TFlops. Similar difference exists for GEMM results.

Results can be accessed as CI artefacts here (including raw summary files from `run_all.sh`): https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/9637082068

**Comment by ESI-SYD** [comment]
1. I can reproduce reported numbers in local IDC (8 cards, MAX 1550) server for both cases with: 
```
pytorch-gpu-dev-0.5.1
gfx-driver-ci-comp_igc-24994
```
2. Also similar data when changing to use custom libigc1_1.0.24994.16243-igc+releaseinternal1_amd64.deb.

3. The CI workflow looks good, not found big difference in dumped ir and asm files between CI data and local ones. 

Does this machine dedicated ? I see there are 2 GPU cards with max frequency 1600, Can you check their health using like: `xpu-smi health -l` or `xpu-smi dump -d 0 -m 0,1,2 -i 1 -n 5`

**Comment by ESI-SYD** [comment]
I think CPU should have limited impact. 

Latest build failed to benchmark gemm and attention (llvm-target): https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/9830987255

But it cannot be reproduced in other branch(like a mirror branch): https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/9833038783

Needs to double confirm with this.

**Comment by ESI-SYD** [comment]
This benchmarks on POC branches will not be merged into the main branch directly, We are integrating them into Triton benchmark like softmax now.

Related:  
https://github.com/intel/intel-xpu-backend-for-triton/pull/1539
https://github.com/intel/intel-xpu-backend-for-triton/pull/1597

**Comment by pbchekin** [comment]
Revisit after integrating to the POC branch to the default one.

**Comment by ESI-SYD** [comment]
Depends on https://github.com/intel/intel-xpu-backend-for-triton/issues/1450 and https://github.com/intel/intel-xpu-backend-for-triton/issues/1451

**Comment by Dewei-Wang-sh** [comment]
since #1450  is closed, please check there should not be perf diff @ESI-SYD @LiyangLingIntel

**Comment by ESI-SYD** [comment]
Update: GEMM done
FlashAttention blocked by it's unavailability on the llvm-target branch. 
```bash
Assertion `detail::isPresent(Val) && "dyn_cast on a non-existent value"' failed.
```
Tracked in: https://github.com/intel/intel-xpu-backend-for-triton/issues/1758

**Comment by ESI-SYD** [comment]
Update:
`Triton`'s FlashAttention wip #1758
`XeTLA`'s FlashAttention done #1877

**Comment by whitneywhtsang** [comment]
Remaining FA performance gap tracked in https://github.com/intel/intel-xpu-backend-for-triton/issues/1758.
